atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/Schema.java

```diff
@@ -177,14 +178,22 @@ public IndexDefinition getIndex(TableReference indexRef) {
      */
     public void validate() {
         // Try converting to metadata to see if any validation logic throws.
-        for (TableDefinition d : tableDefinitions.values()) {
-            d.toTableMetadata();
-            d.getConstraintMetadata();
+        for (Entry<String, TableDefinition> entry : tableDefinitions.entrySet()) {
+            try {
+                entry.getValue().validate();
+            } catch (Exception e) {
+                Throwables.rewrapAndThrowIfInstance("Failed to validate table " + entry.getKey(), e, RuntimeException.class);
```

Comment:
Sorry, yeah, my suggestion was dump. This should just be:

`throw new RuntimeException("Failed to validate table " + entry.getKey(), e);`

Having the logging makes perfect sense - swallowing CheckedExceptions that get added later is going to be sad.